account_asset_number/models/account_asset_profile.py

```diff
@@ -17,7 +17,29 @@ class AccountAssetProfile(models.Model):
         string="Asset Number Sequence",
         domain=lambda self: self._get_domain_sequence_id(),
     )
+    barcode_type = fields.Selection(
+        selection=[("barcode", "Barcode"), ("qr", "QR")],
+        default="barcode",
+    )
+    barcode_width = fields.Integer(
+        compute="_compute_code_size",
+        store=True,
+    )
+    barcode_height = fields.Integer(
+        compute="_compute_code_size",
+        store=True,
+    )
 
     @api.model
     def _get_domain_sequence_id(self):
         return [("company_id", "in", [False, self.env.company.id])]
+
+    @api.depends("barcode_type")
+    def _compute_code_size(self):
```

Comment:
Please, change compute field to onchange.

Step to test:
1. Go to Asset Profile > change Barcode Type
2. change value > Save
3. it will replace with default

change it to onchange for default only

```
@api.onchange("barcode_type")
def _onchange_barcode_type(self):
        # Set default values when type is changed
        if self.barcode_type == "barcode":
            self.barcode_width = 300
            self.barcode_height = 75
        elif self.barcode_type == "qr":
            self.barcode_width = 150
```